### PR TITLE
[MINOR][INFRA] Ensure that docs build successfully with SKIP_API=1

### PR DESCRIPTION
**Pull request by nchammas**
### What changes were proposed in this pull request?

This PR tweaks the docs build so that the general docs are first built with `SKIP_API=1` to ensure that the docs build works without any language being built beforehand.

### Why are the changes needed?

[Committers expect][1] docs to build with `SKIP_API=1` on a fresh checkout. Yet, our CI build does not ensure this. This PR corrects this gap.

[1]: https://github.com/apache/spark/pull/44393/files#r1444169083

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Via test commits against this PR.

[The build now fails][f] if the docs reference an include that has not been generated yet.

[f]: https://github.com/nchammas/spark/actions/runs/7450949388/job/20271048581#step:30:29

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by nchammas** [comment]
cc @HyukjinKwon

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -761,6 +761,9 @@ jobs:
       run: ./dev/lint-r
     - name: Run documentation build
       run: |
+        # Build docs first with SKIP_API to ensure they are buildable without requiring any
+        # language docs to be built beforehand.
+        cd docs; SKIP_API=1 bundle exec jekyll build; cd ..
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:766]: LGTM but just to confirm this doesn't take a lot of time right?

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -761,6 +761,9 @@ jobs:
...
         if [ -f "./dev/is-changed.py" ]; then
           # Skip PySpark and SparkR docs while keeping Scala/Java/SQL docs
           pyspark_modules=`cd dev && python3.9 -c "import sparktestsupport.modules as m; print(','.join(m.name for m in m.all_modules if m.name.startswith('pyspark')))"`
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:769]: or wonder if we should set `SKIP_SCALADOC=1` together with `if [ `./dev/is-changed.py -m ...` condition. That might be better.

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (nchammas) [reply .github/workflows/build_and_test.yml:766]: Correct. This adds around 5 seconds to the "Run documentation build" step.

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (nchammas) [reply .github/workflows/build_and_test.yml:769]: Hmm, could you elaborate a little bit on this? How would this prevent [yesterday's scenario][1]?

[1]: https://github.com/apache/spark/pull/44393#discussion_r1444169083

Comment (HyukjinKwon) [reply .github/workflows/build_and_test.yml:769]: Actually let's just merge this. 5 secs won't affect the elapsed time much anyway.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**